README.md

```diff
@@ -1,2 +1,151 @@
 # jupyterhub-deploy-docker
-Reference deployment of JupyterHub with docker
+
+This repository provides a reference deployment of [JupyterHub](https://github.com/jupyter/jupyterhub), a multi-user [Jupyter Notebook](http://jupyter.org/) environment, on a **single host** using [Docker](https://docs.docker.com).  
+
+This deployment:
+
+* Runs the [JupyterHub components](https://jupyterhub.readthedocs.org/en/latest/getting-started.html#overview) in a Docker container on the host
+* Uses [DockerSpawner](https://github.com/jupyter/dockerspawner) to spawn single-user Jupyter Notebook servers in separate Docker containers on the same host
+* Persists JupyterHub data in a Docker volume on the host
+* Persists user notebook directories in Docker volumes on the host
+* Uses [OAuthenticator](https://github.com/jupyter/oauthenticator) and [GitHub OAuth](https://developer.github.com/v3/oauth/) to authenticate users
+
+![JupyterHub single host Docker deployment](internal/jupyterhub-docker.png)
+
+## Use Cases
+
+Possible use cases for this deployment may include, but are not limited to:
+
+* A JupyterHub demo environment that you can spin up relatively quickly.
+* A multi-user Jupyter Notebook environment for small classes, teams, or departments.
+
+## Disclaimer
+
+This deployment is **NOT** intended for a production environment.  
+
+## Prerequisites
+
+* This deployment uses Docker for all the things. It assumes that you wiil be using [Docker Machine](https://docs.docker.com/machine/overview/) and [Docker Compose](https://docs.docker.com/compose/overview/) on a local workstation or laptop to create, build, and run Docker images on a remote host.  It requires [Docker Toolbox](https://www.docker.com/products/docker-toolbox) 1.11.0 or higher.  See the [installation instructions](https://docs.docker.com/engine/installation/) for your environment.
+* This example configures JupyterHub for HTTPS connections (the default).  As such, you must provide TLS certificate chain and key files to the JupyterHub server.  If you do not have your own certificate chain and key, you can either create self-signed versions, or obtain real ones from [Let's Encrypt](https://letsencrypt.org) (see the [letsencrypt example](examples/letsencrypt/README.md) for instructions).
+
+## Create a Docker Machine
+
+Use [Docker Machine](https://docs.docker.com/machine/) to provision a new host, or to connect to an existing one.   
```

Comment:
I think that the provisioning a docker-machine would be good to put in a separate PR for a doc that gets the user from "Zero to Line 37 of the current README". @jhamrick I agree that we want to rely heavily on external instructions but I think if it's not included that we may be opening up more support/troubleshooting questions. I think it would be reasonable to have "Zero to Line 37" using as much Docker Documentation that we can find (though that has changed rapidly on their site too). Will this work...@jtyberg opens a separate PR with the provisioning stuff he already has as a document or link to it in a new issue. For now, we can let "Zero to Line 37" be as it is written; we can always loop back and make the "true beginner" doc later if we want.